### PR TITLE
 improvement(manager backup feature): Added restoration with restore task

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
Since manager 3.1 now supports restoring backups using restore tasks,
I added a restore task for the backup that is created with test_basic_backup.

I also added a cassandra stress read thread after each of the restorations (both the
manual one and the restore task) to verify that they both of them were completed
successfully.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

# TO BE MERGED ONLY AFTER #5779 IS MERGED

### CODE REVIEW

**Comment by rayakurl** [comment]
@ShlomiBalalis - why is it a draft?

**Review by fgelcer — APPROVED** [review]
LGTM

File: mgmt_cli_test.py [hunk 1]
```diff
@@ -871,6 +891,40 @@ def test_enospc_during_backup(self):
...
+        backup_task = mgr_cluster.create_backup_task(location_list=self.locations)
+        backup_task_status = backup_task.wait_and_get_final_status(timeout=1500)
+        assert backup_task_status == TaskStatus.DONE, \
+            f"Backup task ended in {backup_task_status} instead of {TaskStatus.DONE}"
+        target_node = self.db_cluster.nodes[1]
+        has_enospc_been_reached = False
+        with ignore_no_space_errors(node=target_node):
+            try:
+                reach_enospc_on_node(target_node=target_node)
+                has_enospc_been_reached = True
```
Comment (yarongilor) [inline mgmt_cli_test.py:909]: There's a possible case where, for example, `reach_enospc_on_node` raise an exception after already getting to ENOSPC, but then it won't clean the node.

**Review by yarongilor — APPROVED** [review]
Left one minor comment. LGTM.

**Comment by rayakurl** [comment]
> Left one minor comment. LGTM.

Thanks!

File: mgmt_cli_test.py (hunk 1 above)
Comment (ShlomiBalalis) [reply mgmt_cli_test.py:909]: You're right! No harm in cleaning either way. Fixing.

Comment (ShlomiBalalis) [reply mgmt_cli_test.py]: Fixed, and also removed from the other test function that has used it

**Comment by ShlomiBalalis** [comment]
> @fgelcer , @yarongilor - not sure what happened, but for some reason it seems like you didn't approve. Can you please re-approve?

When I re-push the branch after the reviews, they become stale:
`@ShlomiBalalis ShlomiBalalis dismissed stale reviews from yarongilor and fgelcer via 9d684b3 `

**Review by fgelcer — APPROVED** [review]
LGTM

**Review by yarongilor — APPROVED** [review]
LGTM

**Comment by rayakurl** [comment]
Thanks @fruch !

**Pull request merged**